### PR TITLE
add a `debug_exception` util

### PR DESCRIPTION
**Pull request by uriva**

### CODE REVIEW

File: gamla/debug.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import asyncio
+import builtins
+import logging
+from typing import Text
+
+from toolz import curried
+
+logger = curried.do(logging.info)
```
Comment (erez-work) [inline gamla/debug.py:8]: Please use `side_effect`

File: gamla/debug.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+
+def log_text(text: Text, level: int = logging.INFO):
+    return curried.do(lambda x: logging.log(level, text.format(x)))
```
Comment (erez-work) [inline gamla/debug.py:12]: Same.
Also, why do we need both `log_text` and `logger`?

File: gamla/debug.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+
+do_breakpoint = curried.do(lambda x: builtins.breakpoint())
```
Comment (erez-work) [inline gamla/debug.py:15]: `side_effect`

File: gamla/functional_generic.py [hunk 4]
```diff
@@ -665,11 +665,11 @@ async def do(x):
             await f(x)
             return x
 
-        return do
+    else:
```
Comment (erez-work) [inline gamla/functional_generic.py:668]: Is it a style issue or it won't work without the `else`?

**Review by erez-work — APPROVED** [review]
Awesome, much needed.
I always do this locally:
```
def t(x):
    return x
```

File: gamla/debug.py (hunk 2 above)
Comment (uriva) [reply gamla/debug.py:12]: hmm I can guess that `logger` lets you just use it without giving text

File: gamla/functional_generic.py (hunk 4 above)
Comment (uriva) [reply gamla/functional_generic.py:668]: I get a pre-commit note on it, mypy or something like that

**Pull request merged**